### PR TITLE
docs: add pyarrow example for dataframe getitem and to_dict

### PR DESCRIPTION
**Pull request by V-Alizade**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #1000
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
wonderful, thanks @V-Alizade , looks good on green 🥦

**Comment by MarcoGorelli** [comment]
ah sorry looks like it's not green, you'll also have to overwrite the docstrings in 

https://github.com/narwhals-dev/narwhals/blob/76f4edb93185924d69efb1093ed118e03f2fe7eb/narwhals/stable/v1.py#L154-L176

File: narwhals/stable/v1.py [hunk 1]
```diff
@@ -164,7 +164,8 @@ def to_dict(
         Examples:
             >>> import pandas as pd
             >>> import polars as pl
-            >>> import narwhals.stable.v1 as nw
+            >>> import pyarrow as pa
+            >>> import narwhals as nw
```
Comment (anopsy) [inline narwhals/stable/v1.py:168]: Hey Vahideh, so cool to see you contribution. 

You can leave: import narwhals.stable.v1 as nw
As we're showing here how you can implement stable version of narwhals

Comment (MarcoGorelli) [inline narwhals/stable/v1.py:168]:
```suggestion
            >>> import narwhals.stable.v1 as nw
```

**Pull request merged**